fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx

```diff
@@ -0,0 +1,164 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+
+type Props = {
+  id: number;
+  image?: string;
+  text: string;
+  date: string;
+  lesson_num: number;
+  title: string;
+  author: number;
+};
+
+export const ShortPostcard: React.FC<Props> = ({
+  id,
+  image,
+  text,
+  date,
+  lesson_num,
+  title,
+  author,
+}) => {
+  const formatDate = (dateString: string) => {
+    const options: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    };
+    return new Date(dateString).toLocaleDateString([], options);
+  };
+
+  const [voteUp, setVoteUp] = useState(0);
+  const [voteDown, setVoteDown] = useState(0);
+
+  return (
+    <ShortPostcardWrapper key={id}>
+      <ShortFirstLine>
+        <ShortCardFirstLine>
+          <ShortDate>{formatDate(date)}</ShortDate>
+          <ShortTitle>{title}</ShortTitle>
+        </ShortCardFirstLine>
+        <ShortImageParentDiv>
+          <Image src={image} />
+        </ShortImageParentDiv>
+      </ShortFirstLine>
+      <ShortSecondLine>
+        <ShortLikeDiv>
+          <VoteButton type='button' onClick={() => setVoteUp(voteUp + 1)}>
+            <ShortActionImage
+              alt='like'
+              src={require('../../../images/like-svgrepo-com.svg').default}
+            />
+          </VoteButton>
+          <ShortActionCounter>{voteUp}</ShortActionCounter>
+          <VoteButton type='button' onClick={() => setVoteDown(voteDown + 1)}>
```

Comment:
user state should be checked too: if you already voted up you should `-1` from voteUp state & add `+1` to voteDown

Reply:
added logic

Reply:
same for voting down

Reply:
@svr93